src/activity-window-host.js

```diff
@@ -406,7 +408,10 @@ export class ActivityWindowRedirectHost {
       }
       this.requestId_ = request.requestId;
       this.args_ = request.args;
-      this.returnUrl_ = request.returnUrl;
+      // The "safe" check here is very superficial and only meant to prevent
+      // obvious unsafe URLs. The complete verification is delegated to the
+      // host as part of the `accept()` call.
+      this.returnUrl_ = assertObviousUnsafeUrl(request.returnUrl);
```

Comment:
will something like window.alert() continue to work?

Reply:
@kaipakartik No. In this case it won't work in any situation.